### PR TITLE
DOC: updates documentations Closes #21749

### LINKED ISSUE #21749: DOCS: Update import statements in docstrings
xref https://github.com/pandas-dev/pandas/pull/21731#pullrequestreview-134460126

- [ ] ./pandas/plotting/_converter.py|134 col 17| ">>> from pandas.plotting import register_matplotlib_converters"
- [ ] pandas/plotting/_misc.py|501 col 5| >>> from pandas import read_csv
- [ ] pandas/plotting/_misc.py|502 col 5| >>> from pandas.tools.plotting import parallel_coordinates
- [x] ./pandas/util/testing.py|602 col 5| >>> from pandas.util.testing import capture_stdout https://github.com/pandas-dev/pandas/pull/42979
- [x] ./pandas/util/testing.py|650 col 5| >>> from pandas.util.testing import capture_stderr https://github.com/pandas-dev/pandas/pull/42979
- [x] ./pandas/util/testing.py|2213 col 7| >>> from pandas.util.testing import network https://github.com/pandas-dev/pandas/pull/42979
- [x] ./pandas/util/testing.py|2214 col 7| >>> from pandas.io.common import urlopen https://github.com/pandas-dev/pandas/pull/42979
- [x] ./pandas/tseries/holiday.py|145 col 9| >>> from pandas.tseries.holiday import Holiday, nearest_workday https://github.com/pandas-dev/pandas/pull/45485
- [ ] ./pandas/tseries/holiday.py|146 col 9| >>> from pandas import DateOffset
- [ ] ./pandas/core/sparse/series.py|739 col 9| >>> from numpy import nan
- [ ] pandas/core/frame.py|2844 col 9| >>> from numpy.random import randn
- [ ] pandas/core/frame.py|2845 col 9| >>> from pandas import DataFrame
- [ ] ./pandas/core/algorithms.py|1519 col 5| >>> from pandas.api.extensions import take
- [x] ./pandas/core/dtypes/concat.py|285 col 5| >>> from pandas.api.types import union_categoricals https://github.com/pandas-dev/pandas/pull/45346
- [ ] ./pandas/io/pytables.py|457 col 5| >>> from pandas import DataFrame
- [ ] ./pandas/io/pytables.py|458 col 5| >>> from numpy.random import randn
- [ ] ./pandas/io/json/normalize.py|138 col 5| >>> from pandas.io.json import json_normalize
- [x] pandas/io/stata.py|2713 col 5| >>> from pandas.io.stata import StataWriter117 https://github.com/pandas-dev/pandas/pull/45346

### PR DESCRIPTION
**Pull request by salomondush**
- [x] xref #21749 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_testing/asserters.py [hunk 1]
```diff
@@ -303,10 +303,10 @@ def assert_index_equal(
 
     Examples
     --------
-    >>> from pandas.testing import assert_index_equal
+    >>> import pandas as pd
```
Comment (jreback) [inline pandas/_testing/asserters.py:306]: can you use the standard convntion e.g.

```
from pandas import testing as tm
```

**Review by jreback — CHANGES_REQUESTED** [review]
all of these should use the standard conventions (except for the internal uses of network)

File: pandas/_testing/asserters.py (hunk 1 above)
Comment (salomondush) [reply pandas/_testing/asserters.py:306]: Just to clarify, do you mean using the standard convention on all modified import statements or on this specific line?

Comment (jreback) [reply pandas/_testing/asserters.py:306]: on all

Comment (salomondush) [reply pandas/_testing/asserters.py:306]: Sorry, I am a first-time contributor, but isn't what you're mentioning the same as the original import statements, for example,  `from pandas.io.common import urlopen`?

Comment (jreback) [reply pandas/_testing/asserters.py:306]: no there is a specific convention for testing routines. *only* those that are note user facing testing should use the fully qualified paths

Comment (salomondush) [reply pandas/_testing/asserters.py]: I updated import statements for user-facing testing files and put back fully qualified paths for others. Let me know if that'll work or if more updates are needed. Thank you. I also followed the following line in the contribution guidelines docs.
> Import required libraries (except numpy and pandas)

File: pandas/core/algorithms.py [hunk 2]
```diff
@@ -1462,7 +1462,7 @@ def take(
 
     Examples
     --------
-    >>> from pandas.api.extensions import take
+    >>> from pandas.pai.extensions import take
```
Comment (jreback) [inline pandas/core/algorithms.py:1465]: this is a typo. are we testing these doc-tests?

File: pandas/io/stata.py [hunk 3]
```diff
@@ -3472,7 +3473,8 @@ class StataWriterUTF8(StataWriter117):
 
     Directly write a zip file
     >>> compression = {"method": "zip", "archive_name": "data_file.dta"}
-    >>> writer = StataWriterUTF8('./data_file.zip', data, compression=compression)
+    >>> writer = StataWriterUTF8('./data_file.zip',
```
Comment (jreback) [inline pandas/io/stata.py:3476]: formatting is off here

File: pandas/_testing/asserters.py [hunk 4]
```diff
@@ -1203,17 +1203,17 @@ def assert_frame_equal(
     This example shows comparing two DataFrames that are equal
     but with columns of differing dtypes.
 
-    >>> from pandas._testing import assert_frame_equal
+    >>> from pandas import _testing as ts
```
Comment (jreback) [inline pandas/_testing/asserters.py:1206]: just do these as the standard import

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/algorithms.py (hunk 2 above)
Comment (salomondush) [reply pandas/core/algorithms.py:1465]: No, I think we're only showcasing the functionality of the take method by passing in different keywords.

**Comment by jreback** [comment]
great does this close all of the items in the original issue?

**Review by jreback — APPROVED** [review]

**Comment by salomondush** [comment]
I only changed those in the directory ./pandas/util/testing.py. Others have remained the same.

**Comment by jreback** [comment]
thanks @salomondush happy to have a PR on other items in the original issue as well

**Pull request merged**